src/Files.App/Views/Layouts/BaseLayoutPage.cs

```diff
@@ -903,22 +903,27 @@ private async Task AddShellMenuItemsAsync(List<ContextMenuFlyoutItemViewModel> s
 				}
 			}
 
-			// Add items to main shell submenu
-			mainShellMenuItems.Where(x => x.LoadSubMenuAction is not null).ForEach(async x =>
-			{
-				await x.LoadSubMenuAction();
-
-				ShellContextFlyoutFactory.AddItemsToMainMenu(mainItems, x);
-			});
+			// Filter mainShellMenuItems that have a non-null LoadSubMenuAction
+			var mainItemsWithSubMenu = mainShellMenuItems.Where(x => x.LoadSubMenuAction is not null);
 
-			// Add items to overflow shell submenu
-			overflowShellMenuItems.Where(x => x.LoadSubMenuAction is not null).ForEach(async x =>
+			// Load main submenus asynchronously
+			await Task.WhenAll(mainItemsWithSubMenu.Select(async item =>
```

Comment:
I think `mainItemsWithSubMenu.Select` tasks and `overflowItemsWithSubMenu.Select` tasks can be grouped together and await in one `Task.WhenAll`.